packages/openfeature-js/src/lib/types.ts

```diff
@@ -1,53 +1,99 @@
-export type Context = {
-  userId?: string;
-};
+export type Context = { userId?: string } & Record<string, unknown>;
+
+export type FlagType = 'enabled' | 'boolean' | 'string' | 'number' | 'json';
 
-export type FlagType = 'boolean' | 'string' | 'number' | 'json';
+export interface FlagEvaluationOptions {
+  hooks?: Hook[];
+}
 
 /**
  * This interface is common to both Providers and the SDK presented to the Application Author.
- * This is incidental, and may not continue to be the case, especially as additional configuration is provided. 
+ * This is incidental, and may not continue to be the case, especially as additional configuration is provided.
  */
 export interface Features {
-
   /**
-  * Get a boolean flag value.
-  *
-  * NOTE: In some providers this has distinct behavior from getBooleanValue
-  */
-  isEnabled(flagId: string, defaultValue: boolean, context?: Context): Promise<boolean>;
+   * Get a boolean flag value.
+   *
+   * NOTE: In some providers this has distinct behavior from getBooleanValue
+   */
+  isEnabled(
+    flagId: string,
+    defaultValue: boolean,
+    context?: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<boolean>;
 
   /**
-  * Get a boolean flag value.
-  */
-  getBooleanValue(flagId: string, defaultValue: boolean, context?: Context): Promise<boolean>;
+   * Get a boolean flag value.
+   */
+  getBooleanValue(
+    flagId: string,
+    defaultValue: boolean,
+    context?: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<boolean>;
 
-   /**
+  /**
    * Get a string flag value.
    */
-  getStringValue(flagId: string, defaultValue: string, context?: Context): Promise<string>;
- 
-   /**
+  getStringValue(
+    flagId: string,
+    defaultValue: string,
+    context?: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<string>;
+
+  /**
    * Get a number flag value.
    */
-  getNumberValue(flagId: string, defaultValue: number, context?: Context): Promise<number>;
- 
-   /**
+  getNumberValue(
+    flagId: string,
+    defaultValue: number,
+    context?: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<number>;
+
+  /**
    * Get a object (JSON) flag value.
    */
-  getObjectValue<T extends object>(flagId: string, defaultValue: T, context?: Context): Promise<T>;
+  getObjectValue<T extends object>(
+    flagId: string,
+    defaultValue: T,
+    context?: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<T>;
 }
 
 export interface FeatureProvider extends Features {
   name: string;
 }
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface Client extends Features {}
+// consider this name.
+export interface HasHooks {
+  registerHooks(...hooks: Hook[]): void;
+  get hooks(): Hook[];
+}
+
+// may want to convert this to an interface later.
```

Comment:
What do you mean by this comment?

Reply:
It's from when it was an abstract class.